utils/build-script

```diff
@@ -774,6 +778,115 @@ class BuildScriptInvocation(object):
 
         return options
 
+    def compute_product_classes(self):
```

Comment:
As I mention in a longer comment below, I think it should be a property on HostSpecificConfiguration. For now we'd set every instance to the same parsed set of products.